plugins/plugin-testing-java/plugin-testing-junit/che-plugin-testing-junit-ide/src/main/java/org/eclipse/che/plugin/testing/junit/ide/action/RunAllTestAction.java

```diff
@@ -31,35 +44,63 @@
+                                     implements RunTestActionDelegate.Source {
 
     private final NotificationManager   notificationManager;
     private final TestResultPresenter   presenter;
     private final TestServiceClient     service;
+    private final AppContext            appContext;
+    private final SelectionAgent        selectionAgent;
     private final RunTestActionDelegate delegate;
 
     @Inject
     public RunAllTestAction(JUnitTestResources resources,
-                            NotificationManager notificationManager,
-                            EditorAgent editorAgent,
-                            TestResultPresenter presenter,
-                            FileTypeRegistry fileTypeRegistry,
-                            TestServiceClient service,
-                            JUnitTestLocalizationConstant localization) {
-        super(localization.actionRunAllTitle(), localization.actionRunAllDescription(), resources.testAllIcon(),
-              editorAgent, fileTypeRegistry);
+                                   NotificationManager notificationManager,
+                                   AppContext appContext,
+                                   TestResultPresenter presenter,
+                                   TestServiceClient service,
+                                   SelectionAgent selectionAgent,
+                                   JUnitTestLocalizationConstant localization) {
+        super(Arrays.asList(PROJECT_PERSPECTIVE_ID), localization.actionRunAllTitle(),
+              localization.actionRunAllDescription(), null, resources.testAllIcon());
         this.notificationManager = notificationManager;
-        this.editorAgent = editorAgent;
         this.presenter = presenter;
         this.service = service;
+        this.appContext = appContext;
+        this.selectionAgent = selectionAgent;
         this.delegate = new RunTestActionDelegate(this);
     }
 
     @Override
     public void actionPerformed(ActionEvent e) {
-        Map<String, String> parameters = new HashMap<>();
-        delegate.doRunTests(e, parameters);
+        Resource resource = appContext.getResource();
+        if (resource != null) {
+            Project project = resource.getProject();
+            if (project != null) {
+                Map<String, String> parameters = new HashMap<>();
+                delegate.doRunTests(e, parameters);
+            }
+        }
+    }
+
+    @Override
+    public void updateInPerspective(@NotNull ActionEvent e) {
+        Resource resource = appContext.getResource();
+        if (resource == null) {
+            e.getPresentation().setEnabledAndVisible(false);
+        }
+        
+        Project project = resource.getProject();
+        if (project == null) {
+            e.getPresentation().setEnabledAndVisible(false);
+        }
+        
+        e.getPresentation().setVisible(true);
+
+        String projectType = project.getType();
+        boolean enable = "maven".equals(projectType);
```

Comment:
@vparfonov : because for the moment the Java (JUnit / TestNG) testing feature is only supported on maven Java projects. that was from the beginning of this feature afaik. But sure as soon as this limitation is dropped, we'll be able to remove this test.